src/block-library/post-date/edit.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * Import @wordpress dependencies
+ */
+import { __ } from "@wordpress/i18n";
+import {
+  useBlockProps,
+  InspectorControls,
+  __experimentalDateFormatPicker as DateFormatPicker,
+  __experimentalPublishDateTimePicker as PublishDateTimePicker,
+  BlockControls
+} from "@wordpress/block-editor";
+
+import {
+  PanelBody,
+  ToggleControl,
+  ToolbarGroup,
+  ToolbarButton,
+  Dropdown
+} from "@wordpress/components";
+import { useEntityProp } from "@wordpress/core-data";
+import { useMemo, useState } from "@wordpress/element";
+import { dateI18n, getSettings as getDateSettings } from "@wordpress/date";
+
+/**
+ * Import andt components, dependencies & configuration
+ */
+import { Typography, ConfigProvider } from "antd";
+import { edit } from "@wordpress/icons";
+
+import { updateAttributes, generateStyles } from "../../shared";
+import { textAttributes } from "../../shared/attributes";
+import antdTheme from "../../../../../themes/headless/antd-theme.json";
+import { BlockVisibility } from "../../block-editor/block-visibility";
+import { BlockStyles } from "../../block-editor/block-styles";
+import { TextControls } from "../../shared/controls";
+
+const { Text } = Typography;
+
+/**
+ * Import editor styles
+ */
+import "./editor.scss";
+
+/**
+ * Gutenberg Edit component
+ */
+export default function Edit({
+  attributes,
+  setAttributes,
+  clientId,
+  context: { postId, postType: postTypeSlug, queryId },
+}) {
+  // Used by the Gutenberg editor to save & output blocks properly
+  const blockProps = useBlockProps({
+    className: `gutenberg-ant-design--${clientId}`,
+  });
+
+  // Merge the default attributes with the saved ones
+  const savedAttributes = { ...textAttributes, ...attributes };
+
+  // Component states
+	
+  const isDescendentOfQueryLoop = Number.isFinite(queryId);
+  // used to display date popup that let's the user to edit the date outside Query loop
+  const [popoverAnchor, setPopoverAnchor] = useState(null);
```

Comment:
@Danishkhurshid What pop-up is this referring to? Not seeing it when I play around with the component.